feat(parser): add more helps to parser errors

sapphi-red opened the pull request:
Added more helps to parser errors to make it easier to understand how the error can be resolved.

Boshen commented:
I like this.

Boshen commented:
### Merge activity

* **Oct 31, 7:10 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 31, 7:10 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

sapphi-red committed:
feat(parser): add more helps to parser errors (#15186)

Added more helps to parser errors to make it easier to understand how the error can be resolved.
